base/src/main/java/com/epmedu/animeal/base/ui/AnimealSwitch.kt

```diff
@@ -0,0 +1,171 @@
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.zIndex
+import com.epmedu.animeal.base.R
+import com.epmedu.animeal.base.theme.AnimealTheme
+
+/**
+ * Shows the bar that holds 2 tabs.
+ *
+ * @param onTabSelected Called when the tab is switched.
+ * @param modifier The [Modifier].
+ */
+@Composable
+fun AnimealSwitch(
+    modifier: Modifier = Modifier,
+    onTabSelected: (tab: Tab) -> Unit
+) {
+    var currentTab by remember { mutableStateOf(Tab.Cats) }
+
+    TabRow(
+        selectedTabIndex = currentTab.ordinal,
+        backgroundColor = MaterialTheme.colors.surface,
+        modifier = modifier
+            .size(width = 226.dp, height = 36.dp)
+            .clip(RoundedCornerShape(10.dp)),
+        indicator = { tabPositions ->
+            TabIndicator(tabPositions, currentTab)
+        },
+        divider = {}
+    ) {
+        AnimalTab(
+            tab = Tab.Dogs,
+            selected = currentTab == Tab.Dogs,
+            onClick = {
+                onTabSelected(it)
+                currentTab = it
+            }
+        )
+        AnimalTab(
+            tab = Tab.Cats,
+            selected = currentTab == Tab.Cats,
+            onClick = {
+                onTabSelected(it)
+                currentTab = it
+            }
+        )
+    }
+}
+
+/**
+ * Shows an indicator for the tab.
+ *
+ * @param tabPositions The list of [TabPosition]s from a [TabRow].
+ * @param tab The [Tab] that is currently selected.
+ */
+@Composable
+private fun TabIndicator(
+    tabPositions: List<TabPosition>,
+    tab: Tab
+) {
+
+    val transition = updateTransition(
+        targetState = tab,
+        label = "Tab indicator"
+    )
+
+    val indicatorLeft by transition.animateDp(
+        label = "Indicator left"
+    ) { page ->
+        tabPositions[page.ordinal].left
+    }
+
+    val indicatorRight by transition.animateDp(
+        label = "Indicator right"
+    ) { page ->
+        tabPositions[page.ordinal].right
+    }
+
+    Box(
+        Modifier
+            .wrapContentSize(align = Alignment.BottomStart)
+            .offset(x = indicatorLeft)
+            .width(width = indicatorRight - indicatorLeft)
+            .fillMaxSize()
+            .padding(all = 2.dp)
+            .background(
+                color = MaterialTheme.colors.primary,
+                shape = RoundedCornerShape(9.dp)
+            )
+            .zIndex(zIndex = -1f)
```

Comment:
It's clear that you pass zIndex, you don't need to use named argument here.

Reply:
removed